### PR TITLE
Adapt documentation from "QuantumCitations" fork

### PR DESCRIPTION
**Pull request by goerz**
This makes the documentation appropriate for QuantumCitations having been merged back into DocumenterCitations.

The default style (following QuantumCitations) has changed to `:numeric`, making this a breaking change. The differences to the previous version of DocumenterCitations are outlined in `NEWS.md`, serving as a guideline for any users switching to the next release of DocumenterCitations, but wanting to keep the existing author-year style citations.

### CODE REVIEW

**Comment by goerz** [comment]
I've set `version = "0.3.0-dev"`, in anticipation of this being released as `v0.3.0` (breaking release). There are a few places in the documentation where I explicitly talk about "pre/post v0.3.0", to help any user who wants to switch to the new release, but keep their existing author-year citation style.

**Comment by goerz** [comment]
Good catch in #5… let me rebase this

**Comment by goerz** [comment]
For the record: the easiest way to review the rendered documentation locally is to run `make servedocs` in a checkout of this branch. Or, if you're on Windows or otherwise prefer not to use `make`, run `julia --project=test` and

```
julia> include("devrepl.jl")
julia> servedocs(port=8000, verbose=true)
```

File: Project.toml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 name = "DocumenterCitations"
 uuid = "daee34ce-89f3-4625-b898-19384cb65244"
 authors = ["Michael Goerz <mail@michaelgoerz.net>"]
-version = "0.2.1"
+version = "0.3.0-dev"
```
Comment (fredrikekre) [inline Project.toml:4]:
```suggestion
version = "1.0.0-dev"
```

File: NEWS.md [hunk 2]
```diff
@@ -0,0 +1,11 @@
+# v0.3.0
```
Comment (fredrikekre) [inline NEWS.md:1]: I suggest
```suggestion
# v1.0.0
```

**Comment by fredrikekre** [comment]
From #3, I think the only API thing that needs to be discussed is the return value of `format_citation`, if that is even considered public API. Right now it return either a string, or a vector of markdown segments. In #3 it is updated to be a `MarkdownAST.Paragraph` consistently, but this one is slightly more convoluted to construct as a user who just want to implement a custom style. Perhaps it can return a markdown string consistently and we parse and convert internally?

File: Project.toml (hunk 1 above)
Comment (fredrikekre) [reply Project.toml:4]: We can certainly upgrade to Documenter 1.0 in a non-breaking way. I think it is just mostly very annoying for users if we will have two breaking (according to the Pkg interpretation of SemVer) releases in quick succession.

Comment (goerz) [reply Project.toml:4]: Good point. Alright, I'm good with going to 1.0, then (but I'd still prefer for someone to give this a try in the wild before pushing out the release)

**Comment by goerz** [comment]
> From #3, I think the only API thing that needs to be discussed is the return value of `format_citation`, if that is even considered public API.

At least post-1.0, I would prefer not to consider it (or anything detailed in [Internals](https://juliadocs.org/DocumenterCitations.jl/dev/internals/)) part of the public API. And I agree, we can probably adapt to the next Documenter version without making any breaking changes.

> Right now it return either a string, or a vector of markdown segments. In #3 it is updated to be a `MarkdownAST.Paragraph` consistently, but this one is slightly more convoluted to construct as a user who just want to implement a custom style. Perhaps it can return a markdown string consistently and we parse and convert internally?

Returning a markdown string that then gets parsed and inserted into the output tree is exactly the behavior I would like, see https://github.com/JuliaDocs/DocumenterCitations.jl/issues/6

**Comment by mortenpi** [comment]
I don't have any strong opinions here, and nothing stood out when I skimmed through the PR :slightly_smiling_face: 

The only suggestion I would have is to adopt the [keep a changelog](https://keepachangelog.com/en/1.1.0/) format for the changelog/`NEWS.md`. This would make it consistent with Documenter, and you could potentially benefit from some automated tooling in the future: https://github.com/JuliaDocs/Documenter.jl/blob/master/CHANGELOG.md

File: Project.toml (hunk 1 above)
Comment (goerz) [reply Project.toml]: Ok, I changed any mention of 0.3.0 to 1.0, targeting a 1.0 release in the near future

**Comment by goerz** [comment]
> The only suggestion I would have is to adopt the [keep a changelog](https://keepachangelog.com/en/1.1.0/) format for the changelog/`NEWS.md`.

Sounds good to me! I've changed the format of `NEWS.md` accordingly.

File: Project.toml [hunk 3]
```diff
@@ -1,7 +1,7 @@
 name = "DocumenterCitations"
 uuid = "daee34ce-89f3-4625-b898-19384cb65244"
 authors = ["Michael Goerz <mail@michaelgoerz.net>"]
-version = "0.2.1"
+version = "0.1.0-dev"
```
Comment (fredrikekre) [inline Project.toml:4]:
```suggestion
version = "1.0.0-dev"
```

**Review by fredrikekre — APPROVED** [review]

**Pull request merged**